streams/src/test/java/org/apache/kafka/streams/StreamsBuilderTest.java

```diff
@@ -895,6 +898,55 @@ public void shouldUseSpecifiedNameForAggregateOperationGivenTable() {
             STREAM_OPERATION_NAME);
     }
 
+    @Test
+    public void shouldAllowReadingFromSameTopic() {
+        builder.stream("topic");
+        builder.stream("topic");
+        builder.build();
+    }
```

Comment:
Could you please add a try-catch clause to better document the test?
For example:
```suggestion
    public void shouldAllowReadingFromSameTopic() {
        builder.stream("topic");
        builder.stream("topic");
        
        try {
            builder.build();
        } catch (final TopologyException topologyException) {
            fail("TopologyException not expected");
        }
    }
```
This applies also to the other tests.